File: app/models/category.rb
class Category < ActiveRecord::Base
  has_many :products
  has_many :child_categories, :class_name => "Category", :foreign_key => "parent_id"
  belongs_to :category_group
  belongs_to :parent, :class_name => "Category",
             :foreign_key => "parent_id"

  # Validations
  validates_presence_of :title

  def as_json(options={})
    {
        :id => self.id,
        :name => self.title,
        :hasChildren => self.child_categories.length > 0,
        :parentId => self.parent_id
    }
  end
end

File: client/app/scripts/models/category.js
Frenzli.Category = DS.Model.extend({
    name: DS.attr(),
    parentID: DS.attr(),
    hasChildren: DS.attr()
});

Frenzli.Category.FIXTURES = [{
  id: 1,
  name: 'Тільда',
  hasChildren: true,
  parentID: null
}, {
  id: 2,
  name: 'Скрапбукінг',
  hasChildren: false,
  parentID: null
}, {
  id: 21,
  name: 'Печворк',
  hasChildren: false,
  parentID: 1
}, {
  id: 22,
  name: 'Декупаж',
  hasChildren: true,
  parentID: 1
},{
  id: 23,
  name: 'Полімерна глина',
  hasChildren: false,
  parentID: 22
}]